### ISSUE TITLE
RFC new Architect primitive for ETL

### ISSUE THREAD

**Issue by brianleroux** [architect/architect#807]
WIP!

tl;dr there's a common pattern of using S3 for durable storage of data and fanout pattern for processing it. we can make spinning this pattern up trivial and somewhat standardized so we should. we can build up the prototype as an Architect macro to start.

```
@app
myapp

@etl 
covid19 rate(1 day)
pnw-rainfall rate(1 hour)
```

Each entry in `@etl` would generate the following AWS resources:

- raw data bucket (an S3 bucket that is locked down by default)
- derived data bucket (an S3 bucket for massaged data to land from the extract lambda)
- materialized view bucket (an S3 bucket with the data in its final form)
- extract lambda (like a scheduled function that uses an aws standard schedule expression)
- transform lambda (one or more async functions (step, sns, sqs) that is invoked by the extract lambda)

## open questions

- auth and access 
- upgrade fallback (some jobs take more than the default lambda `/tmp` directory can handle…we could transparently auto upgrade workload into fargate container for those cases)
